src/services/CrawlPageProvince.ts

```diff
@@ -0,0 +1,181 @@
+        const operatorPaging = cheerio.load(resultPaging);
+        operatorPaging(LIST_WAREHOUSES.DOM_URL_WAREHOUSES).each(function () {
+          dataCrawlPathWareHouse.push({
+            url: `${URL_HOME_PAGES}${operatorPaging(this).find('a').attr('href')}`,
+            status: 0,
+          });
+        });
+        fs.writeFile(filePathDetailsWareHouse, JSON.stringify(dataCrawlPathWareHouse), (err) => {
+          if (err) throw err;
+          winston.info(`save url ${optionsPaging.uri} done !`);
+        });
+      }
+    } catch (error) {
+      winston.info(error);
+    }
+  }
+  winston.info('[crawl success data url ware house]');
+  return dataCrawlPathWareHouse;
+}
+
+async function detailPageWarehouse() {
+  try {
+    // Check if the file exists or not
+    if (!fs.existsSync(filePathDetailsWareHouse)) {
+      // Create file;
+      fs.writeFile(filePathDetailsWareHouse, '', (err) => {
+        if (err) throw err;
+      });
+    }
+
+    // Read file json
+    let dataPathWareHouse: any = await readFile(filePathDetailsWareHouse, 'utf-8');
+
+    // Check if the json file pathDetailsWareHouse.json has data, if file no data -> get data in function crawlPathWareHouse()
+    if (Object.keys(dataPathWareHouse).length === 0 || dataPathWareHouse.constructor === Object) {
+      dataPathWareHouse = await crawlPathWareHouse();
+    } else {
+      dataPathWareHouse = JSON.parse(dataPathWareHouse);
+    }
+
+    // Check file json, if status = 0 => crawl data warehouse details with status = 0 -> save data 1 file and change status = 0->1
+    // Check if the file exists or not
+
+    if (!fs.existsSync(pathDetailsWareHouse)) {
+      // Create file;
+      fs.writeFile(pathDetailsWareHouse, '', (err) => {
+        if (err) throw err;
+      });
+    }
+
+    // read file output.json . If data file dataOutput = data file output.json, else dataOutput = []
+    let dataOutput = [];
+    const dataTest = await readFile(pathDetailsWareHouse, 'utf-8');
+    if (Object.keys(dataTest).length === 0 || dataTest.constructor === Object) {
+      dataOutput = [];
+    } else {
+      dataOutput = JSON.parse(dataTest);
+    }
+
+    for (const dataPath of dataPathWareHouse) {
+      const start = Date.now();
+      const dataPage = [];
+      if (dataPath.status === 0) {
+        const optionsProvince = {
+          method: 'GET',
+          uri: dataPath.url,
+        };
+        const resultProvince = await request_promise(optionsProvince);
+        const operator = cheerio.load(resultProvince);
+        const dataImage = [];
+        operator(LIST_DETAILS_WAREHOUSES.DOM_IMAGES).each(function () {
+          dataImage.push(operator(this).find('img').attr('data-src'));
+        });
+        for (let t = 0; t < dataImage.length; t++) {
+          if (dataImage[t]) {
+            dataPage.push({
+              key: `image[${t}]`,
+              value: dataImage[t],
+            });
+          }
+        }
```

Comment:
chỗ này sao xử lí phức tạp vậy ? (2 vòng for)
gôp còn 1 for được k e?

Reply:
.